lib/src/tests/DragAndDrop.spec.tsx

```diff
@@ -0,0 +1,111 @@
+import VernaProvider from ':/providers/VernaProvider';
+import {
+  schemaFactory,
+  translationsFactory,
+  uiSchemaFactory,
+  widgetsFactory,
+} from ':/tests/mocks/factories';
+import ShowCaseWidgetProps from ':/types/Widgets';
+
+describe('drag and drop', () => {
+  it('should display the draggable fields', async () => {
+    function ToolbarWidget(props: ShowCaseWidgetProps) {
+      return <span>{props.widgetName}</span>;
+    }
+
+    render(
+      <VernaToolbar>
+        <ToolbarWidget type="string" widgetName="textwidget" />
+        <ToolbarWidget type="number" widgetName="numberwidget" />
+      </VernaToolbar>,
+    );
+
+    const $numberWidget = screen.getByText('numberwidget').parentElement;
+    expect($numberWidget);
+    expect($numberWidget!.draggable).toBe(true);
+
+    const $TextWidget = screen.getByText('textwidget').parentElement;
+    expect($TextWidget);
+    expect($TextWidget!.draggable).toBe(true);
+  });
+
+  it('should add corresponding field on drop at position', async () => {
+    const translations = translationsFactory();
+
+    render(
+      <Suspense fallback="Loading...">
+        <VernaProvider
+          isEditor
+          DropZone={() => <span>dropzone</span>}
+          defaultSchema={schemaFactory()}
+          defaultUiSchema={uiSchemaFactory()}
+          defaultWidgets={widgetsFactory()}
+          locale="en-US"
+          translations={translations}
+        >
+          <div data-testid="wrapper">
+            <VernaForm />
+          </div>
+        </VernaProvider>
+      </Suspense>,
+    );
+
+    // Wait that the form is rendered...
+    await waitFor(() => {
+      expect(screen.queryByTestId('wrapper')).toBeInTheDocument();
+    });
+
+    // Confirm we start with 2 drop zone
+    let $dropZones = screen.getAllByText('dropzone');
+    expect($dropZones).toHaveLength(2);
+
+    // Simulate the drop of a new widget of type string on the first dropzone
+    fireEvent.drop($dropZones[0], {
+      dataTransfer: {
+        getData: () => 0,
+        items: [
+          {
+            type: 'string',
+            widgetName: 'textWidgetTest',
+          },
+        ],
+        types: ['Object'],
+      },
+    });
+
+    // We should now have 3 drop zone
+    $dropZones = screen.getAllByText('dropzone');
+    expect($dropZones).toHaveLength(3);
+
+    // Check that the two fields are displayed, the new text input and
+    // the initial number field
+    screen.getByRole('textbox', {});
+    screen.getByLabelText('First field');
+
+    // Add a third text field below the number field
+    fireEvent.drop($dropZones[2], {
+      dataTransfer: {
+        getData: () => 0,
+        items: [
+          {
+            type: 'string',
+            widgetName: 'textWidgetTest',
+          },
+        ],
+        types: ['Object'],
+      },
+    });
+
+    // Check that we have the 4 drop zones
+    $dropZones = screen.getAllByText('dropzone');
+    expect($dropZones).toHaveLength(4);
```

Comment:
```suggestion
    expect(screen.getAllByText('dropzone');).toHaveLength(4);
```

Reply:
duplicate of your second comment :stuck_out_tongue: